as.c

```diff
@@ -270,6 +282,91 @@ void assemble_from_file(vm_env *env, const char *filename)
         assemble_line(env, line);
     }
     free(line);
+}
+
+static inline int mem_not_empty(void *p, int len)
```

Comment:
The function strlen() detects continuous non-zero characters and return the length of them.
This function is meant to detect any non-zero characters in a range of buffer, if any is non-zero, the buffer is considered polluted. There is no way to replace this function with strlen().